website/docs/advanced-concepts/access-control/roles.md

```diff
@@ -0,0 +1,247 @@
+
+#### Delete
+
+**Delete** gives the user an access to delete the existing datasources in a workspace. With the delete permission, one can't edit or create a datasource, but, the user has the access to delete it.
+Delete permission can be given at various hierarchical levels - 
+
+##### Workspace
+
+If you give delete permission to the workspace, It gives access to remove any datasource from the respective workspace. 
+
+##### Datasource
+
+**Delete** for a particular datasource allows you to remove that datasource only.
+
+### Groups and roles permissions
+
+For Groups and roles for an Appsmith instance, here are the permissions that can be assigned to user or a group -
+
+#### Create
+
+With this permission, users can create new groups and roles. 
+
+##### Groups
+
+When the create permission is enabled for groups, a user can create a new group. This also automatically enables permission to view, edit, delete, invite, or remove a user from an existing group.
+
+##### Roles
+
+With create permission enables for roles, a user can create a new role. This also automatically enables permission to view, edit, delete, the existing roles.
+
+#### Edit 
+
+With this permission, users can edit the existing groups and roles. 
+
+##### Groups
+
+When the edit permission is enabled for groups, a user can edit an existing group. This also automatically enables permission to view, edit, invite, or remove a user from an existing groups
+
+##### Roles
+
+With edit permission enables for roles, a user can modify an existing role. This also automatically enables permission to edit, view an existing role or assign an associate role.
+
+:::info
+
+You can't edit the built-in roles from Appsmith. You can only the edit the custom roles defined by any user in your instance.
+:::
+
+#### View
+
+With this permission, users can only view the existing groups and roles.
+
+#### Delete
+
+This permission grants users to delete and view the existing groups and roles.
+
+:::info
+
+You can't delete the built-in roles from Appsmith. You can only the delete the custom roles defined by any user in your instance.
+:::
+
+#### Invite Users
+
+This permission grants the user the access to invite users to a group.
+
+#### Remove User
+
+This permission grants the user the access to remove users from a group.
+
+#### Associate Role
+
+With this permission, a user has the access to assign the roles to other users on the instance. Associate role permission restricts a user from creating new roles or, editing, viewing, and deleting the existing ones, but, they can assign the existing roles to various users on the instance.
+
+### Others permissions
+
+In this section, a user can assign permissions for workspaces and Audit logs.
+
+#### Workspace
+
+Workspace level permissions enables a user to perform actions on the workspaces itself. You can create new workspace or edit, delete, view the existing ones.
```

Comment:
in others tab, for existing workspace we have only edit & delete permissions.
CC: @trishaanand